AnkiDroid/src/main/java/com/ichi2/anki/browser/CardBrowserLaunchOptions.kt

```diff
@@ -0,0 +1,53 @@
+/*
+ *  Copyright (c) 2023 David Allison <davidallisongithub@gmail.com>
+ *
+ *  This program is free software; you can redistribute it and/or modify it under
+ *  the terms of the GNU General Public License as published by the Free Software
+ *  Foundation; either version 3 of the License, or (at your option) any later
+ *  version.
+ *
+ *  This program is distributed in the hope that it will be useful, but WITHOUT ANY
+ *  WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+ *  PARTICULAR PURPOSE. See the GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License along with
+ *  this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.ichi2.anki.browser
+
+import android.content.Intent
+import androidx.annotation.CheckResult
+
+/** How the [com.ichi2.anki.CardBrowser] can be launched */
+sealed interface CardBrowserLaunchOptions {
+    data class DeepLink(val search: String) : CardBrowserLaunchOptions
+    data class SearchQueryJs(val search: String, val allDecks: Boolean) : CardBrowserLaunchOptions
+
+    /** Opened from the 'Card Browser' system context menu ([Intent.ACTION_PROCESS_TEXT]) */
+    data class SystemContextMenu(val search: CharSequence) : CardBrowserLaunchOptions
+}
+
+@CheckResult
+fun Intent.toCardBrowserLaunchOptions(): CardBrowserLaunchOptions? {
+    // search card using deep links
+    data?.getQueryParameter("search")?.let {
+        return CardBrowserLaunchOptions.DeepLink(it)
+    }
+
+    // for intent coming from search query js api
+    getStringExtra("search_query")?.let {
+        return CardBrowserLaunchOptions.SearchQueryJs(it, getBooleanExtra("all_decks", false))
+    }
+
+    // Maybe we were called from ACTION_PROCESS_TEXT
+    if (Intent.ACTION_PROCESS_TEXT == action) {
+        val search = getCharSequenceExtra(Intent.EXTRA_PROCESS_TEXT)
+        if (search.isNullOrEmpty()) {
+            return null
+        }
+        return CardBrowserLaunchOptions.SystemContextMenu(search)
+    }
```

Comment:
collided with https://github.com/ankidroid/Anki-Android/commit/02ec4291922ab29632853760d762635398a637fb - I think this is the proper way to mash them together?

```suggestion
    if (Intent.ACTION_PROCESS_TEXT == action) {
        action = Intent.ACTION_DEFAULT
        val search = getCharSequenceExtra(Intent.EXTRA_PROCESS_TEXT)
        if (search.isNullOrEmpty()) {
            return null
        }
        return CardBrowserLaunchOptions.SystemContextMenu(search)
    }```

Reply:
You are the other of both the conflicting PRs, so I'll trust you on the startup issue remaining fixed, just noting for posterity that the one change I no longer see in the final de-conflicted conflict is where the action is bent from `Intent.ACTION_PROCESS_TEXT` to `Intent.ACTION_DEFAULT`

>         action = Intent.ACTION_DEFAULT

...just in case the note is useful for a follow-on